spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -317,21 +381,39 @@ function onChanOpenTry(
   channelIdentifier: Identifier,
   counterpartyPortIdentifier: Identifier,
   counterpartyChannelIdentifier: Identifier,
-  version: string,
-  counterpartyVersion: string) {
+  counterpartyVersion: string) (version: string) {
   // only ordered channels allowed
   abortTransactionUnless(order === ORDERED)
   // validate port ID
   abortTransactionUnless(portIdentifier === "icahost")
   // only allow channels to be created on host chain if the counteparty port ID
   // is in the expected controller portID format.
   abortTransactionUnless(validateControllerPortParams(counterpartyPortIdentifier))
-  // assert that version is expected format `ics27-1.interchain-account-address`
-  abortTransactionUnless(validateVersion(version))
-  // assert that the counterparty version is `ics27-1`  
-  abortTransactionUnless(counterpartyVersion === "ics27-1")
-  // create the interchain account 
-  RegisterInterchainAccount(accAddr, counterpartyPortIdentifier)
+  // create the interchain account with the counterpartyPortIdentifier
+  // and the underlying connectionID on the host chain.
+  address = RegisterInterchainAccount(counterpartyPortIdentifier, connectionID)
+
+  cpMetadata = UnmarshalJSON(counterpartyVersion)
+  abortTransactionUnless(cpMetadata.Version === "ics27-1")
+  // select an encoding and tx type. If none are supported by executing chain then
+  // abortTransaction
+  encoding = SelectEncoding(cpMetadata.Encoding)
+  abortTransactionUnless(encoding != "")
+  txType = SelectTxType(cpMetadata.TxType)
+  abortTransactionUnless(txType != "")
+  abortTransactionUnless(cpMetadata.ControllerConnectionId === counterpartyConnectionId)
+  abortTransactionUnless(cpMetadata.HostConnectionId === connectionId)
```

Comment:
now that the encoding/tx type are a single value, should this be changed to `IsSupported`? It'd be odd for a host chain to select a encoding not requested by a controller